### PR TITLE
Search Tests Sprint 2 automating test cases

### PR DESCRIPTION
**Pull request by Emiliano-Castillo**
Relates to user story https://track.testpro.io/browse/INTERNSHIP-80267

### CODE REVIEW

**Comment by Berichicko** [comment]
Please finish the Internship | Sprint B !You got this problem, please correct it.

image

image

image

image

image

image

File: src/test/java/tests/SearchTests.java [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+    @Test(priority = 7, description = "JIRA Case- 80988. Test User can clear the search query with keyboard esc key and 'x' button", dataProvider = "searchPressingESCKeyData", dataProviderClass = DataProviders.class)
+    public void searchClearingSearchBar (String email, String password, String Song) throws InterruptedException {
+        loginPage = new LoginPage(driver);
+        searchPage = new SearchPage(driver);
+
+        loginPage.userLogin(email, password)
+                .verifyLoginSuccessful();
+        searchPage.searchSongPluto(Song);
+        searchPage.verifyPlutoResultsMatchesSongSectionOnResultsPage();
+        Thread.sleep(1000);
```
Comment (Berichicko) [inline src/test/java/tests/SearchTests.java:88]: Please, delete `throws InterruptedException`

Thread. sleep is bad! It blocks the current thread and renders it unusable for further work.
Using Thread. sleep() frequently in an automation framework is not a good practice.

Please use the Explicit Wait command, the WebDriver is directed to wait until a certain condition occurs before proceeding with executing the code.

File: src/test/java/pageObjects/BasePage.java [hunk 2]
```diff
@@ -15,7 +15,7 @@ public class BasePage {
 
     public BasePage(WebDriver driver) {
         this.driver = driver;
-        wait = new WebDriverWait(driver, Duration.ofSeconds(10));
```
Comment (Berichicko) [inline src/test/java/pageObjects/BasePage.java:18]: HINT

pom

**Comment by Emiliano-Castillo** [comment]
Hey Alex thanks for the feed back please clarify some things here for me

So to my understanding, these tests created here are supposed to fail, aren't they? As an automation engineer aren't we supposed to create the test cases as per requirements? If you take a look at my User Story in JIRA 80267, For example, test case 80982 When the test is executed manually, it fails. For this test case (when one searches for a song/artist/album no matter the search, one should always get results for each section Songs, Artists, and Albums sections should be displayed) when a search is done for artist Grav for example the results show song in the song section, Where its suppose to also display the artist in artist section and album in album section(so this is a bug). SO when creating a test case for automation shouldn't It fail if im creating a test case for what is supposed to be a requirement? 

I don't see the point in creating test cases just so they pass but don't meet requirements you know? 

Also yes I agree with you on the Thread.sleep I will try to fix that test case. I  couldn't get it run as per test requirements because for the test to be valid I would have to wait till this elements are displayed for example. When I ran it with our the Thread,sleep test would pass but it shouldn't pass in this case because once user presses esc key on keyboard the search bar AND results page should also clear and it doesn't but when I took thread.sleep away it would pass as if results page did clear because it wasn't waiting for the proper amount of time. but I will try to fix that.

**Comment by Berichicko** [comment]
Hey Emiliano,

Thanks for sharing your thoughts. I get where you're coming from, and you've raised some solid points.

**Test Failures:**
You're right! When our automated tests fail, it means something's off. Either there's a bug in the system or the test itself needs fixing.

_For example_

Test status: 
Test status_en

	
Test error (red). This status is assigned to tests that caused an exception from the tested source code.
Test error_en



Test failed(yellow). If at least one test receives this status, then all its parents are marked as failed.
Test failed_en


**Automated Test Cases:**
Totally agree – our automated tests should match what the project needs. If a test fails, it's like a red flag saying, "**Hey, something's not right, and we need to look into it**."

**Thread.sleep Usage:**
I hear your concerns about using Thread.sleep. It's not the best move because it makes tests less reliable. Instead, let's use better methods to wait for things to be ready before moving on.


**Test Case Validity:**
Your point about tests being true to real scenarios is spot on. Tests should actually check if the system does what it's supposed to, not just pass for the sake of passing.

**Bug Reporting:**
If you spot something wonky during testing, shout it out! Report it as a bug so the devs can fix it up and keep things on track.

You're doing great. 
Best wishes from the HW team (Alex, [aliaksandr@testpro.io](mailto:aliaksandr@testpro.io))!

**Comment by Emiliano-Castillo** [comment]
Hey Alex, Thanks for the feedback. So In this case do I still get a 50% grade for this sprint?

Screenshot 2024-01-15 at 4 01 03 PM

I went ahead and added an implicit wait just for 1 second to the test method instead. This takes away the need to use Thread.sleep, and test is working as expected. Though, Im afraid I don't think this is the correct approach ethier since I read its not great to mix implicit waits with explicit waits. But its the only way I can think of to get this test case to run as per requirements.

**Comment by Berichicko** [comment]
> Hey Alex, Thanks for the feedback. So In this case do I still get a 50% grade for this sprint?
> 
> Screenshot 2024-01-15 at 4 01 03 PM
> 
> I went ahead and added an implicit wait just for 1 second to the test method instead. This takes away the need to use Thread.sleep, and test is working as expected. Though, Im afraid I don't think this is the correct approach ethier since I read its not great to mix implicit waits with explicit waits. But its the only way I can think of to get this test case to run as per requirements.

OK Please, re-submit Tests Sprint for review

**Pull request merged**